File: 1552-build-an-array-with-stack-operations/1552-build-an-array-with-stack-operations.java
class Solution {
    public List<String> buildArray(int[] target, int n) {
        List<String> res = new ArrayList<>();
        int m=1;
        for(int i : target){
            while (i>m){
                res.add("Push");
                res.add("Pop");
                m+=1;
            }
            res.add("Push");
            m+=1;
        }
        return res;
    }
}

File: 1552-build-an-array-with-stack-operations/1552-build-an-array-with-stack-operations.py
class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        res=[]
        m=1
        for i in target:
            if i>m:res.extend(["Push","Pop"]*(i-m))
            res.append("Push")
            m=i+1
        return res    


    